homeassistant/components/broadlink/light.py

```diff
@@ -0,0 +1,232 @@
+    ATTR_BRIGHTNESS,
+    ATTR_COLOR_TEMP,
+    ATTR_HS_COLOR,
+    COLOR_MODE_COLOR_TEMP,
+    COLOR_MODE_HS,
+    COLOR_MODE_UNKNOWN,
+    SUPPORT_BRIGHTNESS,
+    SUPPORT_COLOR,
+    SUPPORT_COLOR_TEMP,
+    LightEntity,
+)
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+BROADLINK_COLOR_MODE_RGB = 0
+BROADLINK_COLOR_MODE_WHITE = 1
+BROADLINK_COLOR_MODE_SCENES = 2
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Broadlink light."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type == "LB1":
+        lights = [BroadlinkLB1Light(device)]
+
+    async_add_entities(lights)
+
+
+class BroadlinkLight(LightEntity, ABC):
+    """Representation of a Broadlink light."""
+
+    def __init__(self, device):
+        """Initialize the light."""
+        self._device = device
+        self._coordinator = device.update_manager.coordinator
+        self._brightness = round(self._coordinator.data["brightness"] * 2.55)
+        if self._coordinator.data["bulb_colormode"] == BROADLINK_COLOR_MODE_RGB:
+            self._color_mode = COLOR_MODE_HS
+        elif self._coordinator.data["bulb_colormode"] == BROADLINK_COLOR_MODE_WHITE:
+            self._color_mode = COLOR_MODE_COLOR_TEMP
+        else:
+            # Scenes are for now not supported
+            self._color_mode = COLOR_MODE_UNKNOWN
+        self._hs_color = [
+            self._coordinator.data["hue"],
+            self._coordinator.data["saturation"],
+        ]
+        self._color_temp = round(
+            (self._coordinator.data["colortemp"] - 2700) / 100 + 153
+        )
+        self._state = self._coordinator.data["pwr"]
+
+    @property
+    def name(self):
+        """Return the name of the light."""
+        return f"{self._device.name} Light"
+
+    @property
+    def assumed_state(self):
+        """Return True if unable to access real state of the light."""
+        return False
+
+    @property
+    def available(self):
+        """Return True if the light is available."""
+        return self._device.update_manager.available
+
+    @property
+    def is_on(self):
+        """Return True if the light is on."""
+        return self._state
+
+    @property
+    def brightness(self):
+        """Return the brightness of this light between 0..255."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the color property."""
+        return self._hs_color
+
+    @property
+    def color_temp(self):
+        """Return the color temperature property."""
+        return self._color_temp
+
+    @property
+    def should_poll(self):
+        """Return True if the light has to be polled for state."""
+        return False
+
+    @property
+    def supported_features(self):
+        """Return the supported features."""
+        return SUPPORT_BRIGHTNESS | SUPPORT_COLOR | SUPPORT_COLOR_TEMP
```

Comment:
These flags are deprecated in favor of VALID_COLOR_MODES. We can remove this property.

Reference: https://github.com/home-assistant/core/blob/dev/homeassistant/components/light/__init__.py.